testplan/common/utils/table.py

```diff
@@ -14,25 +14,30 @@ class TableEntry(object):
     as a ``list`` of ``list`` or a ``list`` of ``dict``.
     """
 
-    def __init__(self, table):
-        self.table = self._validate_table(table or [])
-
-    @staticmethod
-    def _validate_table(table):
-        if isinstance(table, (list, tuple)):
-            if all_are(table, dict) or all_are(table, list, tuple):
-                return table
-        raise TypeError(
-            (
-                "`table` must be a list of"
-                " lists or list of dicts, got:\n{}".format(table)
+    def __init__(self, table=[]):
+        self.table = None
+        self._is_header_only = False
+        self._is_dict_table = False  # table contains dicts or lists only
+        if isinstance(table, list):
+            tbl_len = len(table)
+            self._is_header_only = tbl_len <= 1
```

Comment:
in case table is a list of dict, this line doesn't hold true.
It could just have one row.